src/exec/handle_redirects.c

```diff
@@ -0,0 +1,96 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   handle_redirects.c                                 :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: reasuke <reasuke@student.42tokyo.jp>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/09/07 17:45:16 by reasuke           #+#    #+#             */
+/*   Updated: 2024/09/07 19:02:54 by reasuke          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "ast.h"
+#include "libft.h"
+#include "utils.h"
+
+static int	handle_redirect_input(const char *filepath)
+{
+	int	fd;
+
+	fd = open(filepath, O_RDONLY | O_CLOEXEC);
+	if (fd == -1)
+	{
+		print_error(filepath, strerror(errno));
+		return (-1);
+	}
+	if (dup2(fd, STDIN_FILENO) == -1)
+	{
+		print_error("dup2", strerror(errno));
+		return (-1);
+	}
+	return (0);
+}
+
+static int	handle_redirect_output(const char *filepath)
+{
+	int	fd;
+
+	fd = open(filepath, O_WRONLY | O_CREAT | O_TRUNC | O_CLOEXEC,
+			S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+	if (fd == -1)
+	{
+		print_error(filepath, strerror(errno));
+		return (-1);
+	}
+	if (dup2(fd, STDOUT_FILENO) == -1)
+	{
+		print_error("dup2", strerror(errno));
+		return (-1);
+	}
+	return (0);
+}
+
+static int	handle_redirects_append(const char *filepath)
+{
+	int	fd;
+
+	fd = open(filepath, O_WRONLY | O_CREAT | O_APPEND | O_CLOEXEC,
+			S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+	if (fd == -1)
+	{
+		print_error(filepath, strerror(errno));
+		return (-1);
+	}
+	if (dup2(fd, STDOUT_FILENO) == -1)
+	{
+		print_error("dup2", strerror(errno));
+		return (-1);
+	}
+	return (0);
+}
+
+int	handle_redirects(t_list *redirects)
+{
+	int	status;
```

Comment:
I believe int is preferred to ssize_t in this context. ssize_t is typically used for representing size values and -1 (for error cases), especially in system calls that return a size or count. In our case, the return value of handle_redirect is strictly limited to 0 (success) or -1 (error). Therefore, using int is sufficient and more appropriate.

Reply:
The return value of handle_redirect currently has only two possible values. This means it could potentially be replaced by a boolean. It will be addressed in issue #51.